### PR TITLE
feat(sdk): init sdk package with biome and a precommit hook

### PR DESCRIPTION
**Pull request by ga-reth**
initialise the SDK package with:

- SDK subdir
- typescript dependency
- compilation / clean steps
- gitignore
- typed abi file
- top level readme

issue: none

### CODE REVIEW

File: sdk/src/utils/package.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "type": "module"
+}
```
Comment (kevinhalliday) [inline sdk/src/utils/package.json:3]: Why do we need these (this and src/types/package.json)?

File: sdk/pnpm-workspace.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+packages:
+  - 'src'
```
Comment (kevinhalliday) [inline sdk/pnpm-workspace.yaml:2]: I'm good with using a workspace but like we said live, might be best just to start with single package until we have need for multiple

**Review by kevinhalliday — APPROVED** [review]
Couple thoughts in comments.

Looks like we've got pre-commit hooks failing. Fix those. And then not would be a good time to introduce some ts precommit / ci hooks. typechecking, linting, formatting, tests

File: sdk/src/utils/package.json (hunk 1 above)
Comment (ga-reth) [reply sdk/src/utils/package.json:3]: for the pnpm workspaces setup, the presence of each `package.json` indicates a new submodule within the workspace

File: sdk/pnpm-workspace.yaml (hunk 2 above)
Comment (ga-reth) [reply sdk/pnpm-workspace.yaml:2]: sounds good, i'll revert to a more simple setup

**Comment by ga-reth** [comment]
hey @kevinhalliday i've dropped the workspaces stuff, and added compilation / clean steps, i'll add precommit stuff in a followup

**Comment by ga-reth** [comment]
@kevinhalliday i've added biome for linting/formatting and a precommit hook

**Pull request merged**